src/audio/pipeline.c

```diff
@@ -1261,8 +1261,14 @@ static int pipeline_xrun_recover(struct pipeline *p)
 /* notify pipeline that this component requires buffers emptied/filled */
 void pipeline_schedule_copy(struct pipeline *p, uint64_t start)
 {
-	if (p->sched_comp->state == COMP_STATE_ACTIVE)
-		schedule_task(&p->pipe_task, start, p->ipc_pipe.deadline);
+	if (p->sched_comp->state == COMP_STATE_ACTIVE) {
+		/* timer driven pipeline should execute task synchronously */
+		if (p->ipc_pipe.timer_delay)
+			pipeline_copy(p->sched_comp);
```

Comment:
This would mean the copy would run at the same IRQ priority as the timer IRQ ? Is there an issue relating to this change ?

Reply:
Running copy at the timer IRQ is the whole point of this change. We want to process all of the pipeline tasks synchronously every system tick. These are all low latency pipelines, so there is no point in running them asynchronously.